app/src/renderer/components/wallet/SendModal.vue

```diff
@@ -0,0 +1,266 @@
+        :disabled="$v.fields.$invalid"
+        value="Send Tokens"
+        color="primary"
+        @click.native="validateForm"
+      />
+    </div>
+  </action-modal>
+</template>
+
+<script>
+import b32 from "scripts/b32"
+import { required, between, integer } from "vuelidate/lib/validators"
+import { mapActions, mapGetters } from "vuex"
+import TmBtn from "common/TmBtn"
+import TmFieldGroup from "common/TmFieldGroup"
+import TmFormGroup from "common/TmFormGroup"
+import TmFormStruct from "common/TmFormStruct"
+import TmPage from "common/TmPage"
+import TmPart from "common/TmPart"
+import TmField from "common/TmField"
+import TmFormMsg from "common/TmFormMsg"
+import TmBalance from "common/TmBalance"
+import FieldAddon from "common/TmFieldAddon"
+import ToolBar from "common/ToolBar"
+import ActionModal from "common/ActionModal"
+
+export default {
+  name: `send-modal`,
+  components: {
+    TmBalance,
+    TmBtn,
+    TmField,
+    FieldAddon,
+    TmFieldGroup,
+    TmFormGroup,
+    TmFormMsg,
+    TmFormStruct,
+    TmPage,
+    TmPart,
+    ToolBar,
+    ActionModal
+  },
+  props: {
+    denom: {
+      type: String,
+      required: true
+    }
+  },
+  data: () => ({
+    bech32error: null,
+    fields: {
+      address: ``,
+      amount: null,
+      denom: ``,
+      password: ``
+    },
+    sending: false
+  }),
+  computed: {
+    ...mapGetters([`wallet`, `connected`]),
+    max() {
+      let denom = this.wallet.balances.find(b => b.denom === this.denom)
+      return (denom && denom.amount) || 0
+    },
+    denominations() {
+      return this.wallet.balances.map(i => ({
+        key: i.denom.toUpperCase(),
+        value: i.denom
+      }))
+    }
+  },
+  mounted() {
+    if (this.denom) {
+      this.fields.denom = this.denom
+    }
+  },
+  methods: {
+    ...mapActions([`sendTx`]),
+    close() {
+      this.$emit(`update:showSendModal`, false)
+    },
+    async validateForm() {
+      this.sending = true
+      this.$v.$touch()
+
+      if (!this.$v.$invalid) {
+        await this.submitForm()
+      } else {
+        this.sending = false
+      }
+    },
+    async submitForm() {
+      const amount = +this.fields.amount
+      const address = this.fields.address
+      const denom = this.fields.denom
+      const type = `send`
+
+      await this.$refs.actionModal.submit(async () => {
+        await this.sendTx({
+          type,
+          password: this.fields.password,
+          to: address,
+          amount: [{ denom, amount: amount.toString() }]
+        })
+
+        this.$store.commit(`notify`, {
+          title: `Successfully Sent`,
+          body: `Successfully sent ${amount} ${denom} to ${address}`
+        })
+      }, `Sending tokens failed`)
+
+      this.sending = false
+    },
+    bech32Validate(param) {
+      try {
+        b32.decode(param)
+        this.bech32error = null
+        return true
+      } catch (error) {
+        this.bech32error = error.message
+        return false
```

Comment:
Minor personal perversion
it could be 
try {
  ..
} catch (e) {
...
} finally {
  return !!this.bech32error
}